core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -978,6 +981,116 @@ class CivilizationInfo {
+            willingness -= 300  // -300 if very recently demanded tribute (10 turns)
+        if (demandingCiv.getDiplomacyManager(this).hasFlag(DiplomacyFlags.RecentlyDemandedTribute))
+            willingness -= 40   // -40 if recently demanded tribute (20 turns)
+        if (getDiplomacyManager(demandingCiv).influence < -30)
+            willingness -= 300  // -300 if less than -30 influence
+
+        val forcerank = gameInfo.getAliveMajorCivs().sortedByDescending { it.getStatForRanking(RankingType.Force) }.indexOf(demandingCiv)
+        willingness += 75 - ((75 / gameInfo.gameParameters.players.size) * forcerank)
+
+        val forcenearcity = getCapital().getCenterTile().getTilesInDistanceRange(1..7)
+            .sumBy { if (it.militaryUnit != null && it.militaryUnit!!.civInfo == demandingCiv)
+                max(it.militaryUnit!!.baseUnit().strength, it.militaryUnit!!.baseUnit().rangedStrength)
+            else 0 }    // Sum up strength (or ranged strength) of units within 7 tiles
+        val csforce = CityCombatant(getCapital()).getCityStrength() + units.sumBy { max(it.baseUnit().strength, it.baseUnit().rangedStrength) }
+        val forceratio = forcenearcity.toFloat() / csforce.toFloat()
+
+        if (forceratio > 3f)
+            willingness += 125
+        else if (forceratio > 2f)
+            willingness += 100
+        else if (forceratio > 1.5f)
+            willingness += 75
+        else if (forceratio > 1f)
+            willingness += 50
+        else if (forceratio > 0.5f)
+            willingness += 25
+
+        println ("Force rank " + forcerank.toString())
+        println ("Force near CS " + forcenearcity.toString() + ", CS Force " + csforce.toString() + ", ratio " + forceratio.toString())
+        println ("Willingness " + willingness.toString())
+
+        return willingness
+    }
+
+    fun goldGainedByTribute(): Int {
+        var gold = when (gameInfo.gameParameters.gameSpeed) {
+            GameSpeed.Quick -> 60
+            GameSpeed.Standard -> 50
+            GameSpeed.Epic -> 35
+            GameSpeed.Marathon -> 30
+        }
+        val turnsToIncrement = when (gameInfo.gameParameters.gameSpeed) {
+            GameSpeed.Quick -> 5f
+            GameSpeed.Standard -> 6.5f
+            GameSpeed.Epic -> 14f
+            GameSpeed.Marathon -> 32f
+        }
+        gold += 5 * (gameInfo.turns / turnsToIncrement).toInt()
+
+        return gold
+    }
+
+    fun demandGold(cityState: CivilizationInfo) {
+        if (!cityState.isCityState()) throw Exception("You can only demand gold from City-States!")
+        val goldAmount = goldGainedByTribute()
+        addGold(goldAmount)
+        cityState.addGold(-goldAmount)
+        cityState.getDiplomacyManager(this).influence -= 15
+        getDiplomacyManager(cityState).setFlag(DiplomacyFlags.RecentlyDemandedTribute, 20)
+        getDiplomacyManager(cityState).setFlag(DiplomacyFlags.VeryRecentlyDemandedTribute, 10)
```

Comment:
Yes, and this is technically wrong according to the second source, but the -300 penalty is insurmountable anyway so it does not change anything.

Reply:
yes, it should not be both, but it does not matter anyway.

Reply:
Let's make it not stack, I'll just add an else if

Reply:
just to clarify because i think what i wrote might be confusing.
-300 from very recently demanded tribute stacks with -300 demanding a worker.
-300 from very recently demanded tribute **does not** stack with -40 recently demanded tribute.
-300 from demanding a worker stacks with -40 for demanding tribute recently.